js/classes/entities/midground/weapon/snowball.js

```diff
@@ -116,8 +116,18 @@ class Snowball {
             this.aliveTimer += GAME.clockTick;
             if (this.aliveTimer >= Snowball.ALIVE_TIME) {
                 this.removeFromWorld = true;
+
+                // spawn a snowball ammo drop on the ground because Chad missed
+                const dropPos = Vector.add(this.pos, new Vector(0, -60));
+                GAME.addEntity(new AmmoDrop(dropPos, AmmoDrop.SNOWBALL));
             }
         }
+
+        if (this.action == "firing" && !this.hasHit && GAME.gameTime % 0.1 < 0.01) {
```

Comment:
Might be nice to put this logic in its own method for readability and same for the other addition of particles in other classes.